peps/pep-9999.rst

```diff
@@ -0,0 +1,557 @@
+and encode the year in the minor version:
+
+* 3.26.0 will be released in 2026
+* 3.27.0 will be released in 2027
+* 3.28.0 will be released in 2028
+* 3.29.0 will be released in 2029
+* 3.30.0 will be released in 2030
+
+For example, 3.26 will be released in 2026.
+It makes it obvious when a release first came out.
+
+Clarity of support lifecycle
+----------------------------
+
+Right now, it’s a little tricky to work out when a release is end-of-life.
+First you have to look up when it was initially released, then add 5 years:
+
+   "When will Python 3.11 be EOL?"
+
+   "Well, let's see... PEP 664 is the 3.11 release schedule, it says 3.11 was
+   released in 2022, EOL after 5 years, so 2022 + 5 = 2027."
+
+But if the initial release year is right there in the version,
+it’s much easier:
+
+    "When will Python 3.26 be EOL?"
+
+    "26 + 5 = [20]31"
+
+Non-goals
+---------
+
+Like the current scheme, only the micro version would be incremented for bug
+fix and security releases, with no change to the major and minor. For example:
+
+.. list-table::
+   :header-rows: 1
+
+   * -
+     - Current scheme
+     - Proposed 3.YY.micro
+   * - Initial release (Oct ’26)
+     - 3.15.0
+     - 3.26.0
+   * - 1st bugfix release (Dec ’26)
+     - 3.15.1
+     - 3.26.1
+   * - 2nd bugfix release (Feb ’27)
+     - 3.15.2
+     - 3.26.2
+   * - ...
+     - ...
+     - ...
+   * - Final security release (Oct ’31)
+     - 3.15.17
+     - 3.26.17
+
+No change to :pep:`602` (Annual Release Cycle for Python):
+
+* No change to the 17 months to develop a feature version: alphas, betas and
+  release candidates.
+
+* No change to the support duration:
+  two years of full support and three years of security fixes.
+
+* No change to the annual October release cadence.
+
+Specification
+=============
+
+Python versions are numbered 3.YY.micro where:
+
+* *3* is the major version number
+  – it is always 3.
+
+* *YY* is the minor version number
+  - it is the short year number: ``{year} - 2000``.
+
+* *micro* is the micro version number
+  - it is incremented for each bugfix or security release.
+
+We'll keep major version 3. Python 3 is the brand, there will be no Python 4.
+
+In the year 2100, the minor will be ``2100-2000 = 100``,
+therefore the version will be 3.100.0.
+
+Python 3.14 is the last version before this change, released in 2025.
+Python 3.26 is the first version after this change, released in 2026.
+There will be no Python 3.15 to 3.25 inclusive.
+
+Backwards compatibility
+=======================
+
+This version change is the safest of the CalVer options considered
+(see `rejected ideas <PEP 9999 rejected_>`_) because we keep major version 3
+and the minor version is still two digits.
+The minor will eventually change to three digits, but this is predictable,
+a long way off and can be planned for.
+
+We retain the ``python3`` executable.
+
+Version mapping
+---------------
+
+Versions 3.15 to 3.25 inclusive will be skipped.
```

Comment:
I wonder whether it would be worth shipping a `python3.15` executable as part of Python 3.26, that just immediately errors out and tells you to use `python3.26` instead

Reply:
This is an interesting idea! I'll mention it as a possibility in "How to teach this" and we can see what folk think.